app-modules/portal/tests/Unit/KnowlegeManagementPortalSearchTest.php

```diff
@@ -0,0 +1,309 @@
+        'description' => 'Focus on the latest developments, research, and innovative ideas shaping the future of education across the globe.',
+    ]);
+
+    KnowledgeBaseItem::factory()->state([
+        'id' => '9dbe944d-330e-40c1-94b2-3312b07a7588',
+        'public' => true,
+        'title' => 'Gamification in Education: Transforming Classrooms into Playgrounds',
+        'article_details' => ['type' => 'doc', 'content' => [['type' => 'paragraph', 'content' => [['type' => 'text', 'text' => 'This article delves into the concept of gamification in education']]]]],
+        'notes' => 'make classrooms more interactive.',
+    ])
+        ->for($knowledgeBaseCategory, 'category')
+        ->create();
+
+    $url = URL::signedRoute(name: 'api.portal.search', absolute: false);
+    $response = json('POST', $url);
+
+    expect($response)->toMatchSnapshot();
+});
+
+test('category should present in response', function () {
+    $settings = app(PortalSettings::class);
+
+    $settings->knowledge_management_portal_enabled = true;
+    $settings->save();
+
+    $knowledgeBaseCategory = KnowledgeBaseCategory::factory()->create([
+        'name' => 'Innovative Learning Approaches',
+    ]);
+
+    $url = URL::signedRoute(name: 'api.portal.search', absolute: false);
+    $response = json('POST', $url);
+
+    $data = $response->json();
+
+    $categoryNames = collect($data['data']['categories'])->pluck('name');
+    expect($categoryNames)->toContain($knowledgeBaseCategory->name)->toMatchSnapshot();
+});
+
+test('filter `featured` articles', function () {
+    $settings = app(PortalSettings::class);
+
+    $settings->knowledge_management_portal_enabled = true;
+    $settings->save();
+
+    $knowledgeBaseCategory = KnowledgeBaseCategory::factory()->state([
+        'name' => 'Tech-Driven Education',
+        'slug' => 'tech-driven-education',
+        'description' => 'Explore how technology is shaping education, from virtual classrooms to AI-driven learning tools.',
+    ])->create();
+
+    KnowledgeBaseItem::factory()->state([
+        'id' => '9dbe944d-330e-40c1-94b2-3312b07a7590',
+        'is_featured' => true,
+        'title' => 'Gamification in Education: Transforming Classrooms into Playgrounds',
+        'category_id' => $knowledgeBaseCategory->getKey(),
+        'public' => true,
+    ])
+        ->create();
+
+    KnowledgeBaseItem::factory()->state([
+        'id' => '9dbe944d-330e-40c1-94b2-3312b07a7557',
+        'is_featured' => false,
+        'title' => 'The Rise of Microlearning: Bite-Sized Education for a Fast-Paced World',
+        'category_id' => $knowledgeBaseCategory->getKey(),
+        'public' => true,
+    ])
+        ->create();
+
+    $url = URL::signedRoute(name: 'api.portal.search', absolute: false);
+    $response = json('POST', $url, ['filter' => 'featured']);
+
+    $response = $response->json();
+
+    expect($response)->data->articles->data->toHaveCount(1)->toMatchSnapshot();
```

Comment:
When asserting that something matches a snapshot, I don't think you need anything like `data->articles->data->toHaveCount(1)` since that is already covered as content in the snapshot